splitgpg2/test_server.py

```diff
@@ -732,6 +732,40 @@ def test_003_option_typo(self) -> None:
             mock.call.warning('Unsupported config option: %s', 'no_such_option'),
         )
 
+    def test_004_multiple_files(self) -> None:
+        reader = mock.Mock()
+        writer = mock.Mock()
+        config = configparser.ConfigParser()
+        tmpdir = tempfile.TemporaryDirectory()
+        confdir = tmpdir.name + '/qubes-split-gpg2'
+        os.makedirs(confdir + '/conf.d', 0o700)
+        os.environ['XDG_CONFIG_HOME'] = tmpdir.name
+        default_file = confdir + '/qubes-split-gpg2.conf'
+        first_extra_file = confdir + '/conf.d/00_test.conf'
+        second_extra_file = confdir + '/conf.d/01_test.conf'
+        with open(default_file, 'w', encoding='utf-8') as d:
+            d.write("""
+            [client:testvm]
+            allow_keygen = yes
+            verbose_notifications = no
+            """)
+        with open(first_extra_file, 'w', encoding='utf-8') as e:
+            e.write("""
+            [client:testvm]
+            verbose_notifications = no
+            """)
+        with open(second_extra_file, 'w', encoding='utf-8') as e:
+            e.write("""
+            [client:testvm]
+            allow_keygen = no
+            verbose_notifications = yes
+            """)
+        config_loaded = load_config_files('testvm')
+        gpg_server = GpgServer(reader, writer, 'testvm')
+        gpg_server.load_config(config_loaded)
+        self.assertTrue(gpg_server.allow_keygen)
```

Comment:
Should this actually be true? The last entry is "no".

Reply:
Sorry if it is not clear. The other files are loaded are the following:

- `/etc/qubes-split-gpg2.conf`
- `$XDG_CONFIG_HOME/qubes-split-gpg2/conf.d/*.conf`
- `$XDG_CONFIG_HOME/qubes-split-gpg2/qubes-split-gpg2.conf`

The `first_extra_file` and `second_extra_file` are in `conf.d/*.conf`, while `default_file` is the last to be sourced, which takes

The order the files are written (`default_file`, `first_extra_file`, `second_extra_file`) are different from the order they are loaded (`first_extra_file`, `second_extra_file`, `default_file`).

Do you think I should reorder the writing of files for easier comprehension?

Note the header on `qubes-split-gpg2.conf.example`:
```
# Place this file in ~/.config/qubes-split-gpg2/qubes-split-gpg2.conf
# It will take precedence over any other included configuration file, such
# as the drop-in '.conf' files in ~/.config/qubes-split-gpg2/conf.d/
```

And now I see I didn't document the same way on the README, fixing with next commit.

Reply:
Usual convention is that conf.d drop-ins override things set in the main config. Do you see some specific reason why doing it differently here?

Reply:
Response by e-mail went to the main thread https://github.com/QubesOS/qubes-app-linux-split-gpg2/pull/13#issuecomment-2119009473

Reply:
Ok, then the test needs some change, because currently it will pass also if no drop-in file is loaded at all...

Reply:
Included more evaluations in the test, considering user config loaded at last, or configuration directories sort and load order.